server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -53,6 +66,11 @@ public ClusterAllocationExplanation(ShardRouting shardRouting, @Nullable Discove
     }
 
     public ClusterAllocationExplanation(StreamInput in) throws IOException {
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
```

Comment:
I assume BwC stands for backwards compatibility, is that correct?  (I've seen it several times already, but I figured I should finally confirm).  

I assume that adding an extra field in the API response is not a breaking change, is that correct?  

I am also assuming we are being extra cautious and not introducing it in 7.x, but we could have technically done so, had we wanted to, is that correct?

Reply:
That's correct, BwC is short for backwards compatibility.

Adding an extra field to the response is indeed not considered a breaking change, we expect clients to skip fields they don't recognise.

The labels on the PR indicate how this will be backported, so yes this will go into 7.x. However we have to start out with the change being 8.0-only because we need it to pass all the mixed-version tests in CI first. We'll then backport it and (simultaneously) adjust the version on this line to match. It's quite a complex dance.

Reply:
Got it, @DaveCTurner, thanks a lot for the context.